### ISSUE TITLE
[xtend-maven-plugin] provide option to suppress "Directory is empty. Can't process."

### ISSUE THREAD

**Issue by enikao** [eclipse/xtext-xtend#787]
My parent pom contains a few directories that may contain sources, and runs `xtend-install-debug-info` on them. They do not contain sources in all sub-poms.

I'd like to get a warning-free build, but Xtend emits lots of warnings like

````
[INFO] --- xtend-maven-plugin:2.17.1:xtend-install-debug-info (default) @ com.example.tests ---                        
[WARNING] Directory C:\projects\bla\com.example.tests\src\main\java is empty. Can't process.
````
So, could we have an option on the maven task to suppress these warnings?

**Comment by cdietrich** [comment]
@enikao could you come up with a pr?

**Comment by enikao** [comment]
I'm looking into it. At the moment, I'm still 300,000 errrors away from a clean workspace.

**Comment by cdietrich** [comment]
@enikao you might need to do a `mvn -f releng/org.eclipse.xtext.dev-bom/ install` in lib to get towards that goal

**Comment by enikao** [comment]
Thanks for the hint, although it didn't change anything.

Maven is not happy either:
````
C:\xtext\git\xtext-umbrella\releng                                                                                                                                                                                                          
$ "\Program Files\Maven\bin\mvn" install                                                                                                                                                                                                                         
[INFO] Scanning for projects...                                                                                                                                                                                                                                  
[INFO] Computing target platform for MavenProject: org.eclipse.xtext.feature:org.eclipse.xtext.docs:2.18.0-SNAPSHOT @ C:\xtext\git\xtext-umbrella\releng\org.eclipse.xtext.docs.feature\pom.xml                                                                  
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-lib/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                         
[INFO] Adding repository https://ci.eclipse.org/xtext/job/xtext-lib/job/master/lastStableBuild/artifact/build/p2-repository                                                                                                                                      
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-core/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                        
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-core/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                        
[INFO] Adding repository https://ci.eclipse.org/xtext/job/xtext-core/job/master/lastStableBuild/artifact/build/p2-repository                                                                                                                                     
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-extras/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                      
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-extras/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                      
[INFO] Adding repository https://ci.eclipse.org/xtext/job/xtext-extras/job/master/lastStableBuild/artifact/build/p2-repository                                                                                                                                   
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-eclipse/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                     
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-eclipse/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                     
[INFO] Adding repository https://ci.eclipse.org/xtext/job/xtext-eclipse/job/master/lastStableBuild/artifact/build/p2-repository                                                                                                                                  
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-xtend/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                       
[INFO] Fetching p2.index from https://ci.eclipse.org/xtext/job/xtext-xtend/job/master/lastStableBuild/artifact/build/p2-repository/ (172B)                                                                                                                       
[INFO] Adding repository https://ci.eclipse.org/xtext/job/xtext-xtend/job/master/lastStableBuild/artifact/build/p2-repository                                                                                                                                    
[INFO] Adding repository https://download.eclipse.org/modeling/tmf/xtext/updates/orbit/2019-06                                                                                                                                                                   
[INFO] Fetching p2.index from https://download.eclipse.org/tools/orbit/S-builds/S20190501151401/repository/                                                                                                                                                      
[INFO] Fetching p2.index from https://download.eclipse.org/tools/orbit/S-builds/S20190501151401/repository/                                                                                                                                                      
[INFO] Fetching p2.index from https://download.eclipse.org/tools/orbit/downloads/drops2/S20190501151401/repository/                                                                                                                                              
[INFO] Fetching p2.index from https://download.eclipse.org/tools/orbit/downloads/drops2/S20190501151401/repository/                                                                                                                                              
[ERROR] Failed to resolve target definition C:\xtext\git\xtext-umbrella\releng\org.eclipse.xtext.sdk.target\org.eclipse.xtext.sdk.target.target: Could not find "org.objectweb.asm/7.0.0.v20181030-2244" in the repositories of the current location -> [Help 1] 
[ERROR]                                                                                                                                                                                                                                                          
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.                                                                                                                                                                              
[ERROR] Re-run Maven using the -X switch to enable full debug logging.                                                                                                                                                                                           
[ERROR]                                                                                                                                                                                                                                                          
[ERROR] For more information about the errors and possible solutions, please read the following articles:                                                                                                                                                        
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException                                                                                                                                                                        
                                                                                                                                                                                                                                                                 ````
At least the last listed orbit repo (https://download.eclipse.org/tools/orbit/downloads/drops2/S20190501151401/repository/plugins/?d) only contains org.objectweb.asm v7.1.

**Comment by cdietrich** [comment]
That is due a breaking orbit change yesterday. We hope to get the prs merged tomorrow 
https://github.com/eclipse/xtext/issues/1440

**Comment by ArneDeutsch** [comment]
Fixed https://github.com/eclipse/xtext/issues/1440 by updating to ASM 7.1. Should be working again with next build.

**Comment by cdietrich** [comment]
Fixed in 2.18
